File: src/dataset/typescript/twelve-days_exercism-wackerow/claude/ClaudeSonnet4_twelve-days_v4.ts
class TwelveDays {
    private static readonly _numbers = [
        'first',
        'second',
        'third',
        'fourth',
        'fifth',
        'sixth',
        'seventh',
        'eighth',
        'ninth',
        'tenth',
        'eleventh',
        'twelfth',
    ] as const

    private static readonly _gifts = [
        'a Partridge in a Pear Tree.\n',
        'two Turtle Doves, ',
        'three French Hens, ',
        'four Calling Birds, ',
        'five Gold Rings, ',
        'six Geese-a-Laying, ',
        'seven Swans-a-Swimming, ',
        'eight Maids-a-Milking, ',
        'nine Ladies Dancing, ',
        'ten Lords-a-Leaping, ',
        'eleven Pipers Piping, ',
        'twelve Drummers Drumming, '
    ] as const

    public static recite(fromVerse: number, toVerse: number): string {
        const verses: string[] = []
        
        for (let i = fromVerse - 1; i < toVerse; i++) {
            const gifts: string[] = []
            
            for (let j = i; j >= 0; j--) {
                if (j === 0 && i > 0) {
                    gifts.push('and ' + this._gifts[0])
                } else {
                    gifts.push(this._gifts[j])
                }
            }
            
            verses.push(`On the ${this._numbers[i]} day of Christmas my true love gave to me: ${gifts.join('')}`)
        }
        
        return verses.join('')
    }
}

export default TwelveDays

File: src/dataset/Java/twelve-days_exercism-shashwata27/claude/ClaudeSonnet4_TwelveDays_v4.java
import java.util.List;

class TwelveDays {
    private static final String[] ORDINALS = {
        "", "first", "second", "third", "fourth", "fifth", "sixth",
        "seventh", "eighth", "ninth", "tenth", "eleventh", "twelfth"
    };
    
    private static final String[] GIFTS = {
        "", "a Partridge in a Pear Tree.", "two Turtle Doves",
        "three French Hens", "four Calling Birds", "five Gold Rings",
        "six Geese-a-Laying", "seven Swans-a-Swimming", "eight Maids-a-Milking",
        "nine Ladies Dancing", "ten Lords-a-Leaping", "eleven Pipers Piping",
        "twelve Drummers Drumming"
    };
    
    private static final String PREFIX = "On the ";
    private static final String MIDDLE = " day of Christmas my true love gave to me: ";
    private static final String AND = "and ";

    String verse(int verseNumber) {
        StringBuilder sb = new StringBuilder(150);
        sb.append(PREFIX).append(ORDINALS[verseNumber]).append(MIDDLE);
        
        if (verseNumber == 1) {
            sb.append(GIFTS[1]).append('\n');
            return sb.toString();
        }
        
        for (int i = verseNumber; i > 1; i--) {
            sb.append(GIFTS[i]).append(", ");
        }
        sb.append(AND).append(GIFTS[1]).append('\n');
        
        return sb.toString();
    }

    String verses(int startVerse, int endVerse) {
        StringBuilder sb = new StringBuilder((endVerse - startVerse + 1) * 200);
        
        for (int i = startVerse; i <= endVerse; i++) {
            sb.append(verse(i));
            if (i != endVerse) {
                sb.append('\n');
            }
        }
        
        return sb.toString();
    }
    
    String sing() {
        return verses(1, 12);
    }
}